### PR TITLE
DFR-2811-12, 2816-17, 2894-7 Update miam sections 2, 3a-d

### PR DESCRIPTION
**Pull request by jamiephillipshmcts**
https://tools.hmcts.net/jira/browse/DFR-2811
https://tools.hmcts.net/jira/browse/DFR-2812
https://tools.hmcts.net/jira/browse/DFR-2816
https://tools.hmcts.net/jira/browse/DFR-2817
https://tools.hmcts.net/jira/browse/DFR-2894
https://tools.hmcts.net/jira/browse/DFR-2895
https://tools.hmcts.net/jira/browse/DFR-2896
https://tools.hmcts.net/jira/browse/DFR-2897

_Update the MIAM sections, 2, 3a-d (Form A and New Paper Case)_

**When merging into Prod need to:**
- Update docmosis form A templates to replace the Base templates with the test AAT templates (remove the Test part of the AAT file name)
- Check end to end tests written

### CODE REVIEW

**Review by jthmcts — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/FinremCaseDetailMapperTest.java [hunk 1]
```diff
@@ -420,39 +422,20 @@ private void assertEnums(FinremCaseData caseData) {
...
-            )
-        ));
-
-        assertEquals(MiamPreviousAttendance.FR_MS_MIAM_PREVIOUS_ATTENDANCE_CHECKLIST_VALUE_2,
-            miamValues.getMiamPreviousAttendanceChecklist());
-
-        assertEquals(MiamOtherGrounds.FR_MS_MIAM_OTHER_GROUNDS_CHECKLIST_VALUE_1, miamValues.getMiamOtherGroundsChecklist());
-
-
+        List<MiamExemption> expectedMiamExemptions = new ArrayList<>(Arrays.asList(MiamExemption.values()));
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/FinremCaseDetailMapperTest.java:425]: `List.of`

**Review by jthmcts — CHANGES_REQUESTED** [review]
Just the one minor change

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/FinremCaseDetailMapperTest.java (hunk 1 above)
Comment (jamiephillipshmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/FinremCaseDetailMapperTest.java]: Thanks for reviewing. Will update this now

**Review by jthmcts — DISMISSED** [review]

**Review by AbuSalamHMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MiamCheckService.java [hunk 2]
```diff
@@ -1,30 +1,84 @@
...
+
+    private List<String> getMiamEvidenceUnavailableErrors(Map<String, Object> caseData, String checklistKey,
+                                                          String textboxKey, String checklistValue) {
+        String checklist = Objects.toString(caseData.get(checklistKey));
+        String textbox = convertObjectToString(caseData.get(textboxKey));
+
+        if (checklist != null && checklist.contains(checklistValue) && StringUtils.isBlank(textbox)) {
+            return List.of(MIAM_EVIDENCE_UNAVAILABLE_ERROR);
+        }
+        return List.of();
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MiamCheckService.java:78]: `Collections.emptyList()`

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MiamCheckService.java [hunk 3]
```diff
@@ -1,30 +1,84 @@
...
 
-    public List<String> miamExemptAttendCheck(CaseDetails caseDetails) {
+    private static final String MIAM_EVIDENCE_UNAVAILABLE_ERROR = "Please explain in the textbox why you are unable to "
+        + "provide the required evidence with your application.";
+
+    public List<String> validateMiamFields(CaseDetails caseDetails) {
         Map<String, Object> caseData = caseDetails.getData();
+
+        List<String> miamExemptionErrors = miamExemptionAttendanceCheck(caseData);
+        if (miamExemptionErrors != null && !miamExemptionErrors.isEmpty()) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MiamCheckService.java:39]: Seems like an unnecessary null check

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MiamCheckService.java (hunk 2 above)
Comment (jamiephillipshmcts) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MiamCheckService.java]: Updated

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MiamCheckService.java (hunk 3 above)
Comment (jamiephillipshmcts) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MiamCheckService.java]: Updated

**Review by jthmcts — APPROVED** [review]

**Review by AbuSalamHMCTS — APPROVED** [review]

**Review by jthmcts — APPROVED** [review]

**Pull request merged**